components/timeout/timeout.js

```diff
@@ -45,40 +47,47 @@ class SessionTimeoutUI {
     this.continueBtn.addEventListener('click', this.handleContinue.bind(this));
     this.saveBtn.addEventListener('click', handleSave);
 
-    dialog.init();
-    dialog.show();
+    dialogModule.dialog.init();
+    dialogModule.dialog.show();
   }
 
   handleContinue (e) {
-    e.preventDefault();
-
-    fetch(SessionTimeoutUI.sessionContinueUrl)
-      .then(() => {
-        dialog.hide();
-        this.continueBtn.reset();
-        this.continueRetryCount = this.continueRetryLimit;
-        this.animation.reset();
-        SessionTimeoutUI.reset();
-        this.continueSuccessCallback();
-      }).catch(() => {
-        // if error retry 5 times
-        if (this.continueRetryCount-- > 0) {
-          window.setTimeout(() => {
-            this.handleContinue(e);
-          }, 1000);
-        } else {
-          this.continueBtn.reset();
-          this.continueRetryCount = this.continueRetryLimit;
-        }
-      });
+    if(e) {
+      e.preventDefault();
+    }
+
+    fetchModule.fetch(SessionTimeoutUI.sessionContinueUrl)
+      .then(this.continueSuccess)
+      .catch(this.continueFail);
+  }
+
+  continueSuccess () {
+    dialogModule.dialog.hide();
+    this.continueBtn.reset();
+    this.continueRetryCount = this.continueRetryLimit;
+    this.animation.reset();
+    SessionTimeoutUI.reset();
+    this.continueSuccessCallback();
+  }
+
+  continueFail () {
+    // if error retry 5 times
+    if (this.continueRetryCount-- > 0) {
```

Comment:
Given that this decrement is a post value change, I'd probably move it into the if block.